### PR TITLE
[NateTV] Add extractor(s)

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

A lot of the code is from https://github.com/ytdl-org/youtube-dl/pull/30330

### CODE REVIEW

File: yt_dlp/extractor/nate.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            'title': '[결승 오프닝 타이틀] 2018 LCK 서머 스플릿 결승전 kt Rolster VS Griffin',
+            'description': 'md5:e1b79a7dcf0d8d586443f11366f50e6f',
+            'thumbnail': 'http://image.pip.cjenm.com/CLIP/GA/B120189687/B120189687_EPI0056_03_B.jpg',
+            'upload_date': '20180908',
+            'age_limit': 15,
+            'duration': 73,
+            'uploader': '2018 LCK 서머 스플릿(롤챔스)',
+            'channel': '2018 LCK 서머 스플릿(롤챔스)',
+            'channel_id': 3606,
+            'uploader_id': 3606,
```
Comment (nao20010128nao) [inline yt_dlp/extractor/nate.py:28]: aren't they expected to be string value? :thinking:

Comment (Ashish0804) [reply yt_dlp/extractor/nate.py:28]: yup.
Already fixed locally.
But i do have doubt regarding `tags`, is it safe to just use `'tags': list` 🤔

**Review by pukkandan — CHANGES_REQUESTED** [review]
can merge once these are addressed

File: yt_dlp/extractor/nate.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        '32': '360p',
+        '31': '270p',
+    }
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        video_data = self._download_json(f'https://tv.nate.com/api/v1/clip/{id}', id)
+        formats = [{
+            'url': f_url,
+            'resolution': self._QUALITY.get(f_url[-2:]),
```
[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/nate.py:71]:
```suggestion
            'height': self._QUALITY.get(f_url[-2:]),
```
and change the _QUALITY values to int. The `resolution` will be auto-generated from it

File: yt_dlp/extractor/nate.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            program_data = self._download_json(f'https://tv.nate.com/api/v1/program/{id}/clip/ranking?size=20&page={page_num}',
+                                               id, note=f'Downloading page {page_num}')
+            for clip in program_data.get('content') or []:
+                clip_id = clip.get('clipSeq')
+                if clip_id:
+                    yield self.url_result(
+                        'https://tv.nate.com/clip/%s' % clip_id,
+                        ie=NateIE.ie_key(), video_id=clip_id)
+            if program_data.get('last'):
+                break
```
Comment (pukkandan) [inline yt_dlp/extractor/nate.py:120]: ideally, this should be a PagedList

**Pull request merged**